tools/src/dbgen/Lib.hs

```diff
@@ -30,38 +35,55 @@ import           Pos.Wallet.Web.ClientTypes
 import           Pos.Wallet.Web.ClientTypes.Instances ()
 import           Pos.Wallet.Web.Methods.Logic
 import           Pos.Wallet.Web.Methods.Restore
+import           Pos.Wallet.Web.State.State           (getWalletUtxo, setWalletUtxo, updateWalletBalancesAndUtxo)
+import           Test.QuickCheck                      (arbitrary, generate)
 import           Rendering
 import           Text.Printf
 
 --
 -- Types
 --
 
-data GenSpec = GenSpec {
-    wallets       :: !Integer
+data GenSpec = GenSpec
+    { wallets     :: !Integer
     -- ^ How many wallets to create
     , wallet_spec :: WalletSpec
     -- ^ The specification for each wallet.
+    , fakeUtxo    :: FakeUtxoSpec
     } deriving (Show, Eq, Generic)
 
 instance Interpret GenSpec
 
-data WalletSpec = WalletSpec {
-    accounts       :: !Integer
+data WalletSpec = WalletSpec
+    { accounts      :: !Integer
     -- ^ How many accounts to generate
-    , account_spec :: AccountSpec
+    , account_spec  :: AccountSpec
     -- ^ How specification for each account.
     } deriving (Show, Eq, Generic)
 
 instance Interpret WalletSpec
 
-data AccountSpec = AccountSpec {
-    addresses :: !Integer
-    -- How many addresses to generate.
+data AccountSpec = AccountSpec
+    { addresses :: !Integer
+    -- ^ How many addresses to generate.
     } deriving (Show, Eq, Generic)
 
 instance Interpret AccountSpec
 
+-- TODO(ks): The question here is whether we need to support some other
+-- strategies for distributing money, like maybe using a fixed amount
+-- of `toAddress` to cap the distribution - we have examples where we
+-- have around 80 000 addresses which is a lot and can be intensive?
+-- For now, KISS.
+data FakeUtxoSpec = FakeUtxoSpec
+    {  fromAddress  :: !Integer
+    -- ^ How many addresses to contain fake ADA.
+    ,  amount       :: !Integer
+    -- ^ How much ADA do we want to distribute in a single address.
```

Comment:
Agree, excellent observation. For now let's add uniform stake distribution, later we can change the dhall config to specify a distribution policy (per wallet, that is).